### PR TITLE
Add `ubsan` test suite and fix resulting errors. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -2849,19 +2897,23 @@ def test_nestedstructs(self):
...
-        Bits *b2 = NULL;
-        printf("*%zu*\\n", sizeof(Bits));
-
+        printf("*%zu*\n", sizeof(Bits));
         return 0;
       }
       '''
     # Bloated memory; same layout as C/C++
-    self.do_run(src, '*16,0,4,8,8,12|20,0,4,4,8,12,12,16|24,0,20,0,4,4,8,12,12,16*\n*0,0,0,1,2,64,68,69,72*\n*2*')
+    self.do_run(src, '*16,0,4,8,8,12|20,0,4,4,8,12,12,16|24,0,20,0,4,4,8,12,12,16*\n*0,0,1,2,64,68,69,72*\n*2*')
```
Comment (kripken) [inline tests/test_core.py:2916]: Why did a result change here?

Comment (sbc100) [reply tests/test_core.py:2916]: I removed a single print statement which was to print the pointer itself.. which was previously a null pointer so the address of zero.

Comment (sbc100) [reply tests/test_core.py:2916]: See above where `B *b = NULL;` and we use to print `long(b)` .. which is clearly always zero.

Comment (kripken) [reply tests/test_core.py:2916]: Ah, I see now, thanks.

**Review by kripken — APPROVED** [review]

**Pull request merged**